RecoJets/JetProducers/src/PileupJetIdAlgo.cc

```diff
@@ -494,17 +494,25 @@ PileupJetIdentifier PileupJetIdAlgo::computeIdVariables(const reco::Jet * jet, f
 	if ( lLeadEm == nullptr )   { lLeadEm   = lTrail; }
 	if ( lLeadCh == nullptr )   { lLeadCh   = lTrail; }
 	
-	internalId_.nCharged_    = pfjet->chargedMultiplicity();
-	internalId_.nNeutrals_   = pfjet->neutralMultiplicity();
-	internalId_.chgEMfrac_   = pfjet->chargedEmEnergy()    /jet->energy();
-	internalId_.neuEMfrac_   = pfjet->neutralEmEnergy()    /jet->energy();
-	internalId_.chgHadrfrac_ = pfjet->chargedHadronEnergy()/jet->energy();
-	internalId_.neuHadrfrac_ = pfjet->neutralHadronEnergy()/jet->energy();
-	internalId_.nParticles_ = jet->numberOfDaughters();
-
+	if( patjet != nullptr ) { // to enable running on MiniAOD slimmedJets
+	 internalId_.nCharged_    = patjet->chargedMultiplicity();
+	 internalId_.nNeutrals_   = patjet->neutralMultiplicity();
+	 internalId_.chgEMfrac_   = patjet->chargedEmEnergy()    /jet->energy();
+	 internalId_.neuEMfrac_   = patjet->neutralEmEnergy()    /jet->energy();
+	 internalId_.chgHadrfrac_ = patjet->chargedHadronEnergy()/jet->energy();
+	 internalId_.neuHadrfrac_ = patjet->neutralHadronEnergy()/jet->energy();
+	} else {
+	 internalId_.nCharged_    = pfjet->chargedMultiplicity();
+	 internalId_.nNeutrals_   = pfjet->neutralMultiplicity();
+	 internalId_.chgEMfrac_   = pfjet->chargedEmEnergy()    /jet->energy();
+	 internalId_.neuEMfrac_   = pfjet->neutralEmEnergy()    /jet->energy();
+	 internalId_.chgHadrfrac_ = pfjet->chargedHadronEnergy()/jet->energy();
+	 internalId_.neuHadrfrac_ = pfjet->neutralHadronEnergy()/jet->energy();
```

Comment:
Is there any chance `jet-energy()` could be zero? Maybe with corrupted data? It would be safer to check first that it is > 0 before dividing by it.